api/handlers/service_instance.go

```diff
@@ -195,14 +195,19 @@ func (h *ServiceInstance) delete(r *http.Request) (*routing.Response, error) {
 
 	serviceInstanceGUID := routing.URLParam(r, "guid")
 
+	payload := new(payloads.ServiceInstanceDelete)
+	if err := h.requestValidator.DecodeAndValidateURLValues(r, payload); err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "Unable to decode request query parameters")
+	}
+
 	serviceInstance, err := h.serviceInstanceRepo.GetServiceInstance(r.Context(), authInfo, serviceInstanceGUID)
```

Comment:
Not sure if it is ideal to get the instance in the handler, since now we need to execute get again in the repo, so we can remove the finalizer if purge is true. But since we check whats the instance type so we can return the proper response, i left it for now. One alternative approach would be to return the instance in DeleteServiceInstance in the repo.

Reply:
Yes, but here https://github.com/cloudfoundry/korifi/blob/7947cc6b4964175f7f822270166c343c1c0c2ba0/api/handlers/service_instance.go#L194 we check the type so we can return the proper response. I guess we can return the service instance type from DeleteServiceInstance :/

Reply:
Your idea of returning the deleted service type so that we don't have to lookup the instance twice sounds good. However it would look weird for a delete operation's signature to return type and error, so why not return the record for the instance just deleted. This way it would be possible to check its type and the signature will be logical